### ISSUE TITLE
When this extension is active, setting a link inside a table cell incorrectly creates a nested table

### ISSUE THREAD

**Issue by Abdull** [netresearch/t3x-rte_ckeditor_image#102]
Using 
* TYPO3 9.5.23
* rte_ckeditor 9.5.23
* rte_ckeditor_image 9.0.5

In the CKEditor, inside an HTML table cell, selecting some text, then clicking "Insert link" and inserting a link (TYPO3 link, e-mail link or external HTTP link) incorrectly creates an additional HTML table.
That additional HTML table is a child of aforementioned HTML table cell and is ancestor to the actual created HTML anchor link.

This error does not happen when `resource: "EXT:rte_ckeditor_image/Configuration/RTE/Plugin.yaml"` is not imported.

I therefore suspect `EXT:rte_ckeditor_image/Resources/Public/JavaScript/Plugins/typo3image.js` to be responsible for this error, in particular the magic happening in the `insertElement` event handler, https://github.com/netresearch/t3x-rte_ckeditor_image/blob/cdb3792e0780c169bba7708b8d78a5574d07f0a2/Resources/Public/JavaScript/Plugins/typo3image.js#L154 .

**Comment by mcmulman** [comment]
Hi @Abdull,

cannot reproduce.
Still a problem?
How does your yml file look like where you import _EXT:rte_ckeditor_image/Configuration/RTE/Plugin.yaml_?

**Comment by CybotTM** [comment]
✅ **Issue Resolved**

This issue was the first report of the table-cell link problem and was eventually fixed in later versions.

**Fix Timeline:**
- Reported: November 2020 (TYPO3 v9.5.23)
- Could not reproduce: December 2020
- Actually fixed: March 2022 (commit 7b3038b, PR #137)
- Fix available in: v11.0.3+

**Related Issues:**
- #136 - Same issue for v10 (October 2021)
- #182 - Same issue for v11 (July 2022, confirmed fixed in v11.0.4)

**Technical Cause:**
The `insertElement` event handler in typo3image.js was interfering with link creation inside table cells, causing nested table creation.

**Fix:**
Added guard clause to skip processing when selected element is a `<td>`:
```javascript
if (selectedElement && selectedElement.getName() === 'td'){
    return;
}
```

**Current Status:**
- v10: ❌ Never fixed
- v11.0.3+: ✅ Fixed
- v12/v13: Not affected (CKEditor 5 architecture)
